dbos/_queue.py

```diff
@@ -78,6 +80,12 @@ def enqueue(
             dbos_logger.warning(
                 f"Priority is not enabled for queue {self.name}. Setting priority will not have any effect."
             )
+        if self.partition_queue and (
```

Comment:
Also print a warning/throw an exception if the queue is not configured to be partitioned, but the user provided a partition key?